### PR TITLE
Upgrade run_trial.py and  architect.py

### PR DESCRIPTION
**Pull request by sifa1024**
**What this PR does / why we need it**:
If I use the original program, I will get this error when running darts-gpu, 

```
<architect.Architect object at 0x7fe597aad780>
Traceback (most recent call last):
  File "/home/sifa/docker/katib/examples/v1beta1/trial-images/darts-cnn-cifar10/run_trial.py", line 259, in <module>
    main()
  File "/home/sifa/docker/katib/examples/v1beta1/trial-images/darts-cnn-cifar10/run_trial.py", line 155, in main
    train(train_loader, valid_loader, model, architect, w_optim, alpha_optim,
  File "/home/sifa/docker/katib/examples/v1beta1/trial-images/darts-cnn-cifar10/run_trial.py", line 194, in train
    architect.unrolled_backward(train_x, train_y, valid_x, valid_y, lr, w_optim)
  File "/home/sifa/docker/katib/examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py", line 69, in unrolled_backward
    self.virtual_step(train_x, train_y, xi, w_optim)
  File "/home/sifa/docker/katib/examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py", line 56, in virtual_step
    vw.copy_(w - torch.FloatTensor(xi) * (m + g + self.w_weight_decay * w))
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```

**Which issue(s) this PR fixes** 

None. I've create pull request directly.

**Checklist:**

- [x] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
@sifa1024 I apologize for delaying the response.

@kubeflow/wg-automl-leads Could you approve CI?

File: examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def __init__(self, model, w_momentum, w_weight_decay):
         self.w_momentum = w_momentum
         self.w_weight_decay = w_weight_decay
 
-    def virtual_step(self, train_x, train_y, xi, w_optim):
+    def virtual_step(self, train_x, train_y, xi, w_optim, device):
```
[the remark this answers is not in the document]
Comment (sifa1024) [reply examples/v1beta1/trial-images/darts-cnn-cifar10/architect.py:29]: Yes we should do like this

**Comment by tenzen-y** [comment]
@sifa1024 Could you rebase this PR since we have fixed the CI errors here: https://github.com/kubeflow/katib/pull/2267.

**Comment by sifa1024** [comment]
@tenzen-y I modify code and commits and I rebased this PR.

**Comment by tenzen-y** [comment]
@sifa1024 Could you sign to the DCO?

**Comment by tenzen-y** [comment]
@sifa1024 I guess that you need to sign to all commits with `git commit -s`.

**Comment by sifa1024** [comment]
sorry I am first time use this....

**Comment by tenzen-y** [comment]
> sorry I am first time use this....

@sifa1024 Uhm, could you create a separate PR? It seems that unintended commits are included in this PR.

**Comment by sifa1024** [comment]
I'm going to turn this into a draft. I feel like I messed up.

**Comment by sifa1024** [comment]
@tenzen-y Could I creat a new PR?

**Comment by tenzen-y** [comment]
> @tenzen-y Could I creat a new PR?

Sure.

**Pull request closed without merging**